python/service/service.py

```diff
@@ -519,23 +523,31 @@ def respond_to_test_content_properties(self, event):
         reported_property_test = test_content.get(Fields.REPORTED_PROPERTY_TEST)
 
         for property_name in reported_property_test:
-            if property_name.startswith("prop_"):
+            if device_data and property_name.startswith("prop_"):
                 property_value = reported_property_test[property_name]
 
                 with device_data.reported_property_list_lock:
                     if property_value:
                         operation_id = property_value[Fields.ADD_OPERATION_ID]
-                        device_data.reported_property_values[property_name] = property_value
+                        if Fields.REMOVE_OPERATION_ID in property_value:
+                            device_data.reported_property_values[property_name] = property_value
                     else:
-                        operation_id = device_data.reported_property_values[property_name][
+                        if (
```

Comment:
Why can't this be an `elif`? Having an additional nesting here is less clean imo

Reply:
I'll make a note to revisit that code.  I agree that it's confusing and I think it needs something more than de-nesting.  I'm just not sure what. :)